src/common/secret.c

```diff
@@ -101,8 +101,8 @@ int get_secret_option(const char *secret, const char *key_name,
 
   option[olen] = '\0';
 
-
-  if (secret) {
+  /* when parsing kernel options (-o remount) we get '<hidden>' as the secret */
+  if (secret && strncmp(secret, "<hidden>", 8)) {
```

Comment:
Why not `strcmp(...) != 0`? I strongly prefer explicit return comparisons.

Reply:
Sure, no problem.  I'll update the PR, thanks.